tests/test_requests_batch.py

```diff
@@ -74,6 +89,108 @@ def test_code():
 
 
+@pytest.mark.aiohttp
+@pytest.mark.asyncio
+async def test_requests_query_auto_batch_enabled(
+    event_loop, aiohttp_server, run_sync_test
+):
+    from aiohttp import web
+    from gql.transport.requests import RequestsHTTPTransport
+
+    async def handler(request):
+        return web.Response(
+            text=query1_server_answer_list,
+            content_type="application/json",
+            headers={"dummy": "test1234"},
+        )
+
+    app = web.Application()
+    app.router.add_route("POST", "/", handler)
+    server = await aiohttp_server(app)
+
+    url = server.make_url("/")
+
+    def test_code():
+        transport = RequestsHTTPTransport(url=url)
+
+        with Client(
+            transport=transport,
+            batch_interval=0.01,
+        ) as session:
+
+            query = gql(query1_str)
+
+            # Execute query synchronously
+            result = session.execute(query)
+
+            continents = result["continents"]
+
+            africa = continents[0]
+
+            assert africa["code"] == "AF"
+
+            # Checking response headers are saved in the transport
+            assert hasattr(transport, "response_headers")
+            assert isinstance(transport.response_headers, Mapping)
+            assert transport.response_headers["dummy"] == "test1234"
+
+    await run_sync_test(event_loop, server, test_code)
+
+
+@pytest.mark.aiohttp
+@pytest.mark.asyncio
+async def test_requests_query_auto_batch_enabled_two_requests(
+    event_loop, aiohttp_server, run_sync_test
+):
+    from aiohttp import web
+    from gql.transport.requests import RequestsHTTPTransport
+    from threading import Thread
+
+    async def handler(request):
+        return web.Response(
+            text=query1_server_answer_twice_list,
+            content_type="application/json",
+            headers={"dummy": "test1234"},
+        )
+
+    app = web.Application()
+    app.router.add_route("POST", "/", handler)
+    server = await aiohttp_server(app)
+
+    url = server.make_url("/")
+
+    def test_code():
+        transport = RequestsHTTPTransport(url=url)
+
+        with Client(
+            transport=transport,
+            batch_interval=0.01,
+        ) as session:
+
+            def test_thread():
+                query = gql(query1_str)
+
+                # Execute query synchronously
+                result = session.execute(query)
+
+                continents = result["continents"]
+
+                africa = continents[0]
+
+                assert africa["code"] == "AF"
+
+                # Checking response headers are saved in the transport
+                assert hasattr(transport, "response_headers")
+                assert isinstance(transport.response_headers, Mapping)
+                assert transport.response_headers["dummy"] == "test1234"
+
+            for _ in range(2):
+                thread = Thread(target=test_thread)
+                thread.start()
```

Comment:
should we wait for the thread?

Reply:
I don't think it's really necessary but it does not hurt, so I changed it.